### ISSUE TITLE
hasFormat false positive

### ISSUE THREAD

**Issue by djereg** [briannesbitt/Carbon#2185]
Hello,

I encountered an issue with the following code:
```php
echo Carbon::hasFormat('2020-09-09', 'Y.m.d');
```

Carbon version: 2.39.1

PHP version: 7.4.9

I expected to get:

```
false
```

But I actually get:

```
true
```

For this format, the generated regex is `([1-9]?[0-9]{4}).(1[012]|0[1-9]).(3[01]|[12][0-9]|0[1-9])` where the problem is the `.` between the parts, instead of `\.`, which will pass everything on that places.

```php
Carbon::hasFormat('2020-09-09', 'Y.m.d'); // true
Carbon::hasFormat('2020.09.09', 'Y.m.d'); // true
Carbon::hasFormat('2020*09*09', 'Y.m.d'); // true
Carbon::hasFormat('2020k09d09', 'Y.m.d'); // true
```

https://github.com/briannesbitt/Carbon/blob/7af467873250583cc967a59ee9df29fabab193c1/src/Carbon/Traits/Comparison.php#L859-L887

Thanks!

**Comment by kylekatarnls** [comment]
I confirm this regressed from 2.39.0 to 2.39.1.

**Comment by kylekatarnls** [comment]
Fixed in 2.39.2. Thanks for the report.
